jcl/source/windows/JclTD32.pas

```diff
@@ -1391,6 +1397,44 @@ procedure TJclTD32InfoParser.AnalyseUnknownSubSection(const pSubsection: Pointer
   // do nothing
 end;
 
+function TJclTD32InfoParser.FormatProcName(const ProcName: string): string;
+var
+  pchSecondAt, P: PChar;
+begin
+  Result := ProcName;
+  if (Length(ProcName) > 0) and (ProcName[1] = '@') then
+  begin
+    pchSecondAt := StrScan(PChar(Copy(ProcName, 2, Length(ProcName) - 1)), '@');
+    if pchSecondAt <> nil then
+    begin
+      Inc(pchSecondAt);
+      Result := pchSecondAt;
+      P := PChar(Result);
+      while P^ <> #0 do
+      begin
+        if (pchSecondAt^ = '@') and ((pchSecondAt - 1)^ <> '@') then
+          P^ := '.';
+        Inc(P);
+        Inc(pchSecondAt);
+      end;
+    end;
+  end;
+
+  if ( PeIsNameMangled( Result ) <> umNotMangled ) then
+    Result := PeBorUnmangleName( Result );
+end;
+
+procedure TJclTD32InfoParser.GenerateUnmangledNames;
+var
+  i : Integer;
+begin
+  if ( FUnmangledNames.Count <> 0 ) then
+    Exit;
+//  FUnmangledNames.Capacity := NameCount;
+  for i := 0 to NameCount-1 do
+    FUnmangledNames.Add( FormatProcName( UTF8ToString( PAnsiChar( FNames.Items[ i ] ) ) ) );
```

Comment:
No space between parenthesis
respect casing
Please test with older Delphi versions (6/7), I'm definitely not sure they use UTF8 nor that the UTF8ToString method is available